x-pack/plugins/observability_solution/infra/common/http_api/custom_dashboards_api.ts

```diff
@@ -17,11 +17,11 @@ const PayloadRT = rt.type({
   dashboardFilterAssetIdEnabled: rt.boolean,
 });
 
-const ExistingIdRT = rt.partial({
+const ExistingIdRT = rt.type({
   id: rt.string,
 });
 
-const InfraCustomDashboardRT = rt.intersection([AssetTypeRT, PayloadRT]);
+const InfraCustomDashboardRT = rt.intersection([AssetTypeRT, PayloadRT, ExistingIdRT]);
 
 const SavedObjectIdRT = rt.type({
   savedObjectId: rt.string,
```

Comment:
I wonder if we need this `ExistingIdRT` and `SavedObjectIdRT`. Is it possible to represent the `id` in a single type?

Reply:
Yes, now that I changed the endpoints and the `id` is not optional we can have a single type. I used the `SavedObjectIdRT` name and `id` as value